sdk/textanalytics/ai-text-analytics/src/util.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { FullOperationResponse, OperationOptions } from "@azure/core-client";
+import { LanguageDetectionInput, TextDocumentInput } from "./generated";
+import { LroResponse } from "@azure/core-lro";
+import { TextAnalysisOperationOptions } from "./models";
+import { logger } from "./logger";
+
+interface IdObject {
+  id: string;
+}
+
+/**
+ * Given a sorted array of input objects (with a unique ID) and an unsorted array of results,
+ * return a sorted array of results.
+ *
+ * @internal
+ * @param sortedArray - An array of entries sorted by `id`
+ * @param unsortedArray - An array of entries that contain `id` but are not sorted
+ */
+export function sortResponseIdObjects<T extends IdObject, U extends IdObject>(
+  sortedArray: T[],
+  unsortedArray: U[]
+): U[] {
+  const unsortedMap = new Map<string, U>();
+  for (const item of unsortedArray) {
+    unsortedMap.set(item.id, item);
+  }
+
+  if (unsortedArray.length !== sortedArray.length) {
+    const ordinal = unsortedArray.length > sortedArray.length ? "more" : "fewer";
+    logger.warning(
+      `The service returned ${ordinal} responses than inputs. Some errors may be treated as fatal.`
+    );
+  }
+
+  const result: U[] = [];
+  for (const sortedItem of sortedArray) {
+    const item = unsortedMap.get(sortedItem.id);
+    if (item) {
+      result.push(item);
+    }
+  }
+  return result;
+}
+
+/**
+ * @internal
+ */
+export interface AssessmentIndex {
+  document: number;
+  sentence: number;
+  assessment: number;
+}
+
+/**
+ * @internal
+ */
+export function parseAssessmentIndex(pointer: string): AssessmentIndex {
+  const regex = new RegExp(/#\/documents\/(\d+)\/sentences\/(\d+)\/assessments\/(\d+)/);
+  const res = regex.exec(pointer);
+  if (res !== null) {
+    const assessmentIndex: AssessmentIndex = {
+      document: parseInt(res[1]),
+      sentence: parseInt(res[2]),
+      assessment: parseInt(res[3]),
+    };
+    return assessmentIndex;
+  } else {
+    throw new Error(`Pointer "${pointer}" is not a valid Assessment pointer`);
+  }
+}
+
+/**
+ * Parses the index of the healthcare entity from a JSON pointer.
+ * @param pointer - a JSON pointer representing an entity
+ * @internal
+ */
+export function parseHealthcareEntityIndex(pointer: string): number {
+  const regex = new RegExp(/#\/results\/documents\/(\d+)\/entities\/(\d+)/);
+  const res = regex.exec(pointer);
+  if (res !== null) {
+    return parseInt(res[2]);
+  } else {
+    throw new Error(`Pointer "${pointer}" is not a valid healthcare entity pointer`);
+  }
+}
+
+/**
+ * Set the opinion mining property
+ * @internal
+ */
+export function setOpinionMining<X extends { includeOpinionMining?: boolean }>(
+  x: X
+): X & { opinionMining?: boolean } {
+  return { ...x, opinionMining: x.includeOpinionMining };
```

Comment:
NIT: This results in an object that has both `includeOpinionMining` and `opinionMining`. Is that what you want, or is that OK for the application?

Reply:
That function was not used so I deleted it.

Reply:
My favorite kind of solution.